docs/reference/settings/security-settings.asciidoc

```diff
@@ -224,6 +219,35 @@ The hashing algorithm that is used for the
 in-memory cached API key credentials. For possible values, see <<cache-hash-algo>>.
 Defaults to `ssha256`.
 
+[[api-key-service-settings-delete-retention-period]]
+`xpack.security.authc.api_key.delete.retention_period`::
+(<<dynamic-cluster-setting,Dynamic>>)
+Invalidated or expired API keys older than the retention period are eligible for deletion.
+Defaults to `7d`.
+
+--
+NOTE:  The large real-time clock inaccuracies across the cluster nodes can cause problems
+with evaluating the API key retention period (in the likely case that the node performing
+the invalidation is different from the one performing the deletion).
```

Comment:
Nitpicky changes to the wording here:

```suggestion
NOTE:  Large real-time clock inconsistency across cluster nodes can cause problems
with evaluating the API key retention period. That is, if the clock on the node 
invalidating the API key is significantly different than the one performing the deletion,
the key may be retained for longer or shorter than the configured retention period.
```

I think that's a bit clearer? Writing this well is tricky.